src/3devoUpdate/DeviceList.cs

```diff
@@ -0,0 +1,156 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Management;
+using System.Text.RegularExpressions;
+
+namespace devoUpdate {
+  public class USBDeviceList {
+    //public static Guid GUID_DEVINTERFACE_USB_3DEVO = new Guid("4bb54eac-7b4e-472d-8400-4b2101b8833e");
+    public static Guid GUID_USB_CLASS_BUS_DEVICES = new Guid("36fc9e60-c465-11cf-8056-444553540000");
+    public static Guid GUID_USB_CLASS_USB_DEVICE = new Guid("88bae032-5a81-49f0-bc3d-a4ff138216d6");
+    public const UInt16 HARDWARE_VENDOR_ID_3DEVO = 0x16D0;
+    public const UInt16 HARDWARE_PRODUCT_ID_FE = 0x0C5B;
+    public const UInt16 HARDWARE_PRODUCT_ID_GD = 0x0F44;
+    public const UInt16 HARDWARE_VENDOR_ID_ST_GENERIC = 0x0483;
+    public const UInt16 HARDWARE_PRODUCT_ID_ST_GENERIC = 0xDF11;
+    public const string PORT_HARDWARE_ID_FE = "USB\\VID_16D0&PID_0C5B";
+
+    public USBDeviceList( string _DeviceId, string _PnpDeviceId, string _Description ) {
+      this.ClassGuid = "";
+      this.Description = _Description;
+      this.DeviceId = _DeviceId;
+      this.Name = "";
+      this.PnpDeviceId = _PnpDeviceId;
+      this.SerialNumber = 0x0;
+      this.VendorId = 0x0;
+      this.ProductId = 0x0;
+      this.MachineName = MachineType.None;
+    }
+
+    public string ClassGuid { get; set; }
+    public string Description { get; set; }
+    public string DeviceId { get; set; }
+    public string Name { get; set; }
+    public string PnpDeviceId { get; set; }
+    public UInt64 SerialNumber { get; set; }
+    public MachineType MachineName { get; set; }
+    public UInt16 VendorId { get; set; }
+    public UInt16 ProductId { get; set; }
+
+    public enum MachineType : ushort {
+      None = 0,
+      StBootloader,
+      FilamentMaker,
+      AiridDryer,
+      EndOfList,
+    }
+
+    public static void GetSerialDevices( List<USBDeviceList> devicelist) {
+      string SearchQuery = "SELECT * FROM Win32_SerialPort";
+      ManagementObjectSearcher Searcher = new ManagementObjectSearcher(SearchQuery); // In default scope
+
+      foreach( ManagementObject queryObj in Searcher.Get() ) {
+        string pnpDeviceId = (string)queryObj["PNPDeviceID"];
+        Console.WriteLine("Description  : {0}", queryObj["Description"]);
+        Console.WriteLine(" PNPDeviceID : {0}", pnpDeviceId);
+
+        if( string.IsNullOrEmpty(pnpDeviceId) )
+          continue;
+
+        string SearchQueryPNP = "SELECT * FROM win32_PNPEntity where DeviceID='" + pnpDeviceId.Replace("\\", "\\\\") + "'";
+        ManagementObjectSearcher deviceSearch = new ManagementObjectSearcher(SearchQueryPNP); // In default scope
+
+        foreach( ManagementObject device in deviceSearch.Get() ) {
+          string[] hardwareIds = (string[])device["HardWareID"];
+
+          if( (hardwareIds != null) && (hardwareIds.Length > 0) ) {
+            string FilterProduct = $"USB\\VID_{HARDWARE_VENDOR_ID_3DEVO:X4}&PID_{HARDWARE_PRODUCT_ID_FE:X4}";

```

Comment:
For consistency, it would be nicer to not hardcode these VID/PID here, but just pass them as arguments to `getSerialDevices`.

Alternatively, `GetDfuDevices` and `GetSerialDevices` could maybe be merged, since serial USB devices should also show up in `Win32_USBControllerDevice` and it might be useful to also show the COM port for an Airid device (might help disambiguate between multiple attached ones).